### ISSUE TITLE
itr != structs.end(): Unknown struct when testing dice

### ISSUE THREAD

**Issue by gaoqiaizhanglan** [EOSIO/eos#2533]
cleos push action dice deposit '[ "alice", "100.0000 EOS" ]' -p alice:
main.cpp:1181                 main                 ] Failed with error: Assert Exception (10)
itr != structs.end(): Unknown struct 
Failed to serialize type

**Comment by abourget** [comment]
Check if there is code in the contract. This is usually because there's no ABI.

We should have an explicit assertion with proper message here.

**Comment by dzfish** [comment]
same problem 
I have checked the abi using   "cloes get code  dice.abi -a  dice.abi" and it exists.
I run it from empty blockchain data, does it depend on some other contract?

**Comment by learnforpractice** [comment]
deposit not in the "actions" part of tic_tac_toe.abi
```
  "actions": [{
      "name": "create",
      "type": "create",
      "ricardian_contract": ""
    },{
      "name": "restart",
      "type": "restart",
      "ricardian_contract": ""
    },{
      "name": "close",
      "type": "close",
      "ricardian_contract": ""
    },{
      "name": "move",
      "type": "move",
      "ricardian_contract": ""
    }
  ]
```

**Comment by dzfish** [comment]
we are talking about dice, not tic_tac_toe

**Comment by dzfish** [comment]
I update to the lastest version and it works

**Comment by larryk85** [comment]
This was a known issue that was fixed with pull request #2385.
